internal/server/server.go

```diff
@@ -579,6 +595,29 @@ func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.Handler
 	}
 }
 
+func (s *Server) ValidateRequest(nextHandler echo.HandlerFunc) echo.HandlerFunc {
+	return func(ctx echo.Context) error {
+		request := ctx.Request()
+
+		route, params, err := s.router.FindRoute(request)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, err)
+		}
+
+		requestValidationInput := &openapi3filter.RequestValidationInput{
+			Request:    request,
+			PathParams: params,
+			Route:      route,
+		}
+
+		context := context.TODO()
+		if err := openapi3filter.ValidateRequest(context, requestValidationInput); err != nil {
```

Comment:
I think we can just `ctx` here instead of creating a new context.

Reply:
I believe ctx is the wrong type - ValidateRequest takes context.Context and ctx is echo.context

Reply:
Ah, you're right, we need to get it from the request. `request.Context()` should return the right thing.